benchmarks/benchmarks/sperf/combine_regions.py

```diff
@@ -0,0 +1,257 @@
+
+        NOTE: various test classes override + extend this.
+
+        """
+
+        # Load source cubes (full-mesh and regions)
+        with PARSE_UGRID_ON_LOAD.context():
+            self.full_mesh_cube = load_cube(
+                self._parametrised_cache_filename(n_cubesphere, "meshcube")
+            )
+            self.region_cubes = load(
+                self._parametrised_cache_filename(n_cubesphere, "regioncubes")
+            )
+
+        # Remove all var-names from loaded cubes, which can otherwise cause
+        # problems.  Also implement 'imaginary' data.
+        for cube in self.region_cubes + [self.full_mesh_cube]:
+            cube.var_name = None
+            for coord in cube.coords():
+                coord.var_name = None
+            if imaginary_data:
+                # Replace cube data (lazy file data) with 'imaginary' data.
+                # This has the same lazy-array attributes, but is allocated by
+                # creating chunks on demand instead of loading from file.
+                data = cube.lazy_data()
+                data = da.zeros(
+                    data.shape, dtype=data.dtype, chunks=data.chunksize
+                )
+                cube.data = data
+
+        if create_result_cube:
+            self.recombined_cube = self.recombine()
+
+        # Fix dask usage mode for all the subsequent performance tests.
+        self.fix_dask_settings()
+
+    def fix_dask_settings(self):
+        """
+        Fix "standard" dask behaviour for time+space testing.
+
+        Currently this is single-threaded mode, with known chunksize,
+        which is optimised for space saving so we can test largest data.
+
+        """
+
+        import dask.config as dcfg
+
+        # Use single-threaded, to avoid process-switching costs and minimise memory usage.
+        # N.B. generally may be slower, but use less memory ?
+        dcfg.set(scheduler="single-threaded")
+        # Configure iris._lazy_data.as_lazy_data to aim for 100Mb chunks
+        dcfg.set({"array.chunk-size": "128Mib"})
+
+    def recombine(self):
+        # A handy general shorthand for the main "combine" operation.
+        result = recombine_submeshes(
+            self.full_mesh_cube,
+            self.region_cubes,
+            index_coord_name="i_mesh_face",
+        )
+        return result
+
+
+@on_demand_benchmark
+class CreateCube(Mixin):
+    """
+    Time+memory costs of creating a combined-regions cube.
+
+    The result is lazy, and we don't do the actual calculation.
+
+    """
+
+    def setup(
+        self, n_cubesphere, imaginary_data=True, create_result_cube=False
+    ):
+        # In this case only, do *not* create the result cube.
+        # That is the operation we want to test.
+        super().setup(n_cubesphere, imaginary_data, create_result_cube)
+
+    def time_create_combined_cube(self, n_cubesphere):
+        self.recombine()
+
+    def track_addedmem_create_combined_cube(self, n_cubesphere):
```

Comment:
Can we not use our magic decorator approach on these too? #magicdecorator
It seems not to be used in this file.
I think we are also anyway no longer labelling these track results with a 'Mb' unit -- though we were, where this code originally came from in the 'metrics' repo

Reply:
bebf01b157d82bbdcff32b3b3b80179f03865a83